lib/push.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var request = require('request')
+
+var ERR_NO_PUSH_CALLBACK = 'No Push Callback'
+
+var LOG_UPDATE_INCREMENT_SEND = 'push.send'
+var LOG_UPDATE_INCREMENT_SUCCESS = 'push.success'
+var LOG_UPDATE_INCREMENT_FAIL = 'push.failed'
+var LOG_UPDATE_INCREMENT_NO_CALLBACK = 'push.no_push_callback'
+var LOG_OP_NOTIFY_UPDATE = 'push.notifyUpdate'
+
+module.exports = function (log, db) {
+  return {
+    /**
+     *  Notifies all devices that there was an update to the account
+     *
+     * @param uid
+     * @promise
+     */
+    notifyUpdate: function notifyUpdate(uid) {
+      return db.devices(uid).then(
+        function (devices) {
+          log.trace({op: LOG_OP_NOTIFY_UPDATE, devices: devices})
```

Comment:
This will log a lot of into about the devices, including e.g. the push sending/encryption keys.  Possibly easier to understand if it just logs the device ids, esp. since we're logging the more detailed information as we use it below?